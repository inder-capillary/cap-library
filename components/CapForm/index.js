/**
*
* CapForm
*
*/

import React from 'react';
import './_capForm.scss';
import PropTypes from 'prop-types';
import { Form } from "antd";
const classNames = require('classnames');
class CapForm extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {className, children, ...rest} = this.props;
    return (
      <Form {...rest} className={classNames("cap-form", className)}>
        {React.Children.toArray(children)}
      </Form>
    );
  }
}

CapForm.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
};

export default CapForm;
