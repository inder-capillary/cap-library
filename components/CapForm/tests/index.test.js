
import React from 'react';
import { shallow } from 'enzyme';
import { Form } from 'antd';

import CapForm from '../index';

describe('<CapForm />', () => {
  const wrapper = shallow(<CapForm.CapForm />);

  it('Should render antd form', () => {
    expect(wrapper.find(Form)).toHaveLength(1);
  });
});
