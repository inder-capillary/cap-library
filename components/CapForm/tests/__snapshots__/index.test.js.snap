// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapForm /> CapForm renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal cap-form"
  onSubmit={[Function]}
/>
`;
